### PR TITLE
☑️ Rework verify phrase page to fix issues w/ duplicates

### LINKED ISSUE #1008: Verification process with duplicate words in recovery phrase
### Discord Discussion Link

https://discord.com/channels/808358975287722045/888500174685614090/944256044748652585

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

This user wanted to set up the wallet with a new seed phrase

### What did not work?

The user created a new 24 word seed, that turned out to contain 2 times a duplicate word. This is perfectly fine, but the verification process couldn't be completed.

"When it came to verify, I was only given 6 words to select, rather than the 8 needed to verify
the two words that were doubled were needed within the verify section""

More details: https://discord.com/channels/808358975287722045/888500174685614090/944256044748652585

"When I went back to Tally, I had to click create new wallet again, but this time I wasn't asked for a password, but instead instantly given the 24 seed phrase to write down. After writing it down, I noticed that word 4 == word 24, and word 7 == word 9, so two words were doubled up. I headed to the verify section...

I believe that when the verify section first appeared, there were 8 words to select at the bottom, corresponding to the 8 blank spaces above. What was odd however, is that I was asked to provide word number 4 and 7 twice. i.e. It asked for word 1, then word 4, then word 4 again, then word 7, then word 7 again, then word X etc...

When I selected word 4 the first time, it caused both of the word 4 tiles to disappear, meaning I was one short. The same thing happened when it came to selecting the first word 7, I lost both the tiles representing word 7, meaning I was now missing 2 tiles in total. "

Root cause discussed here:
https://discord.com/channels/808358975287722045/888500174685614090/944302752337649694
https://discord.com/channels/808358975287722045/888500174685614090/944304207941492907

### Version

Community Edition v0.11.0

### Relevant log output

_No response_

### LINKED ISSUE #1073: Recovery phrase verification requests same word twice while actual phrase had no duplicates
### Discord Discussion Link

https://discord.com/channels/808358975287722045/888500174685614090/948509616390688768

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

The user tried to create a new 24 word recovery phrase through Tally

### What did not work?

This one may be difficult to reproduce. Extension was already removed so no logs available.

The verification process showed requested two times word number 2. The actual recovery phrase had 24 unique words. 

Word #2 was also only displayed once on the bottom in the orange boxes. There were a total of 8 words on the bottom, so  one word that didn't match any of the numbers above. 

The user stopped the verification process and created a new phrase, that worked.

image

### Version

Community Edition v0.11.2

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by henryboldi**
This PR reworks the verify phrase page's logic to fix issues with duplicate
words. Several users got stuck on the verify phrase page because the interface
assumed that words in the mnemonic were unique, which isn't the case. So now
the logic is based on the index of the words, rather than the words themselves.

To test, fire up the verify phrase page many times, and see if there's anything
funky. In addition, verify that incorrectly ordered selections and correctly
ordered selections correspond to success and failure states.

Closes #1008, closes #1073

### CODE REVIEW

File: ui/pages/Onboarding/OnboardingVerifySeed.tsx [hunk 1]
```diff
@@ -63,72 +63,81 @@ function SuccessMessage({ mnemonic }: { mnemonic: string[] }) {
...
-        (sortedIndexesOfRandomOrderedMnemonicPart &&
-          sortedIndexesOfRandomOrderedMnemonicPart[i]) ||
-        0
-      if (mnemonicToVerify && mnemonicToVerify[assignedNumber] !== word) {
+
+    selectedInOrder.forEach((selectedIndex, i) => {
+      if (sortedIndexesOfRandomOrderedMnemonicPart[i] !== selectedIndex) {
         result = false
       }
     })
```
Comment (Shadowfiend) [inline ui/pages/Onboarding/OnboardingVerifySeed.tsx:102]: Could this all just be one call to `some`?

**Review by Shadowfiend — COMMENTED** [review]
Couple of quick thoughts.

File: ui/pages/Onboarding/OnboardingVerifySeed.tsx (hunk 1 above)
Comment (henryboldi) [reply ui/pages/Onboarding/OnboardingVerifySeed.tsx:102]: Oh right... haha. Thank you

Comment (henryboldi) [reply ui/pages/Onboarding/OnboardingVerifySeed.tsx]: Smart

**Comment by 0xDaedalus** [comment]
Not sure if we want to tackle this problem in this PR or not but one thing I'm noticing (both in this PR and on main) is that removing addresses shifts up your entire stack of previously selected addresses - does not seem desirable:

https://user-images.githubusercontent.com/94649004/164314360-7a834c48-010e-4855-950b-9d664cdabc39.mov

**Comment by henryboldi** [comment]
@0xDaedalus, great point! I'll improve that UX issue in a follow-up PR. I think it's separate

**Review by greg-nagy — APPROVED** [review]
loving this 

let's 🚢

**Pull request merged**